libcudacxx/include/cuda/std/detail/libcxx/include/complex

```diff
@@ -1229,6 +1572,30 @@ arg(_Tp __re)
     return _CUDA_VSTD::atan2f(0.F, __re);
 }
 
+#ifdef __cuda_std__
+template <class _Tp>
+inline _LIBCUDACXX_INLINE_VISIBILITY
+__enable_if_t<
+    is_same<_Tp, __half>::value,
+    __half
+>
```

Comment:
We could improve this via
```suggestion
__enable_if_t<_LIBCUDACXX_TRAIT(is_same, _Tp, __half), __half>
```

But I see that the whole file is a mess still. Lets move this to a followup PR